nasa-gibs/worldview
# Add spotlight capability to tour

## Description

Fixes #3075.

* Integrate React Joyride
* Allow configuration via tour steps
* Add ability to flag tours as "mock" so they don't show in the tour list to users
* Add mock tour to run tests against and serve as example config

## How to test

1. Load the app via this URL to load the mock tour: http://localhost:3000/?tr=hurricane_dorian_september_joyride&mockTour=true
2. Advance to a tour step with a Joyride beacon showing
3. Click the beacon to open the associated tooltip
4. Confirm closing the tooltip moves to the next step
5. Click the next beacon to open the next tooltip
6. Click through next/prev to confirm navigation between steps
7. Advance to the last step and click "last"
8. Confirm tooltip closes and no beacons showing
9. Advance to mock tour step 4
10. Resize browser window and confirm beacon location on map updates 
11. Click beacon to open tooltip
12. Confirm hovering mouse over the spotlight area allows mouse interactions to be picked up by the map
13. Resize browser window and confirm spotlight area on map updates location
